typed-racket-lib/typed-racket/private/type-contract.rkt

```diff
@@ -450,8 +465,12 @@
             (apply or/sc (append other-scs (map t->sc (nbits->base-types nbits)))))]
        [(? Union? t)
         (match (normalize-type t)
-          [(Union: (? Bottom?) elems) (apply or/sc (map t->sc elems))]
-          [(Union: base elems) (apply or/sc (t->sc base) (map t->sc elems))]
+          [(Union-all: elems)
+           (define-values [hash-elems other-elems] (partition hash/chaperone? elems))
+           (define maybe-hash/sc (hash-types->sc hash-elems))
+           (if maybe-hash/sc
+             (apply or/sc maybe-hash/sc (map t->sc other-elems))
```

Comment:
This is the new behavior in `type->contract` that I'd like to test.

`(U #f (Immutable-HashTable String String) (Mutable-HashTable String String))` should make the contract `(or/c #f (hash/c (or string? string?) (or string? string?)))`


(Maybe this is the wrong approach? Maybe `type->contract` should be very simple, then have a `simplify-contract` function?)